File: backend/src/main/java/com/example/smartabsence/model/Seance.java
package com.example.smartabsence.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.time.LocalTime;
import lombok.Data;

@Entity
@Table(name = "seances")
@Data
public class Seance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate date;
    private LocalTime heureDebut;
    private LocalTime heureFin;

    @ManyToOne
    @JoinColumn(name = "enseignant_id")
    private Enseignant enseignant;

    @ManyToOne
    @JoinColumn(name = "classe_id")
    private Classe classe;

    @ManyToOne
    @JoinColumn(name = "matiere_id")
    private Matiere matiere;
}

File: mobile/types/seance.ts
import type { Classe } from './classe';
import type { Matiere } from './matiere';

export type Seance = {
  id: number;
  date: string;
  heureDebut: string;
  heureFin: string;
  enseignant?: { id: number };
  classe: Classe;
  matiere: Matiere;
};
